core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/DefaultAwsCrtV4aHttpSigner.java

```diff
@@ -58,6 +104,81 @@ private static Duration validateExpirationDuration(Duration expirationDuration)
         return expirationDuration;
     }
 
+    private static AwsSigningConfig signingConfig(
+        SignRequest<?, ? extends AwsCredentialsIdentity> request,
+        V4aProperties v4aProperties) {
+
+        AuthLocation authLocation = request.requireProperty(AUTH_LOCATION, AuthLocation.HEADER);
+        Duration expirationDuration = request.property(EXPIRATION_DURATION);
+        boolean isPayloadSigning = request.requireProperty(PAYLOAD_SIGNING_ENABLED, true);
+        boolean isChunkEncoding = request.requireProperty(CHUNK_ENCODING_ENABLED, false);
+        boolean isTrailing = hasTrailer(request.request());
+
+        AwsSigningConfig signingConfig = new AwsSigningConfig();
+        signingConfig.setCredentials(toCredentials(v4aProperties.getCredentials()));
+        signingConfig.setService(v4aProperties.getCredentialScope().getService());
+        signingConfig.setRegion(v4aProperties.getCredentialScope().getRegion());
+        signingConfig.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4_ASYMMETRIC);
+        signingConfig.setTime(v4aProperties.getCredentialScope().getInstant().toEpochMilli());
+        signingConfig.setUseDoubleUriEncode(v4aProperties.shouldDoubleUrlEncode());
+        signingConfig.setShouldNormalizeUriPath(v4aProperties.shouldNormalizePath());
+        signingConfig.setSignedBodyHeader(AwsSigningConfig.AwsSignedBodyHeaderType.X_AMZ_CONTENT_SHA256);
+
+        switch (authLocation) {
+            case HEADER:
+                signingConfig.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+                if (request.hasProperty(EXPIRATION_DURATION)) {
+                    throw new UnsupportedOperationException(
+                        String.format("%s is not supported for %s.", EXPIRATION_DURATION, AuthLocation.HEADER)
+                    );
+                }
+                break;
+            case QUERY_STRING:
+                signingConfig.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_QUERY_PARAMS);
+                if (request.hasProperty(EXPIRATION_DURATION)) {
+                    signingConfig.setExpirationInSeconds(validateExpirationDuration(expirationDuration).getSeconds());
+                }
+                break;
+            default:
+                throw new UnsupportedOperationException("Unknown auth-location: " + authLocation);
+        }
+
+        if (isPayloadSigning) {
+            configurePayloadSigning(signingConfig, isChunkEncoding, isTrailing);
+        } else {
+            configureUnsignedPayload(signingConfig, isChunkEncoding, isTrailing);
+        }
+
+        return signingConfig;
+    }
+
+    private static void configureUnsignedPayload(AwsSigningConfig signingConfig, boolean isChunkEncoding, boolean isTrailing) {
+        if (isChunkEncoding) {
+            if (isTrailing) {
+                signingConfig.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.STREAMING_UNSIGNED_PAYLOAD_TRAILER);
```

Comment:
nit: can we use static import? `signingConfig.setSignedBodyValue(STREAMING_UNSIGNED_PAYLOAD_TRAILER)`. Same as others